core/src/main/java/org/apache/iceberg/BaseEntriesTable.java

```diff
@@ -192,37 +188,46 @@ public <T> Boolean notNaN(BoundReference<T> ref) {
 
       @Override
       public <T> Boolean lt(BoundReference<T> ref, Literal<T> lit) {
-        return compareDateFileContent(ref, lit, compareResult -> compareResult < 0);
+        return ROWS_MIGHT_MATCH;
       }
 
       @Override
       public <T> Boolean ltEq(BoundReference<T> ref, Literal<T> lit) {
-        return compareDateFileContent(ref, lit, compareResult -> compareResult <= 0);
+        return ROWS_MIGHT_MATCH;
       }
 
       @Override
       public <T> Boolean gt(BoundReference<T> ref, Literal<T> lit) {
-        return compareDateFileContent(ref, lit, compareResult -> compareResult > 0);
+        return ROWS_MIGHT_MATCH;
       }
 
       @Override
       public <T> Boolean gtEq(BoundReference<T> ref, Literal<T> lit) {
-        return compareDateFileContent(ref, lit, compareResult -> compareResult >= 0);
+        return ROWS_MIGHT_MATCH;
       }
 
       @Override
       public <T> Boolean eq(BoundReference<T> ref, Literal<T> lit) {
-        return compareDateFileContent(ref, lit, compareResult -> compareResult == 0);
+        if (fileContent(ref)) {
+          Literal<Integer> intLit = lit.to(Types.IntegerType.get());
+          Integer fileContentId = intLit.value();
+          if (fileContentId == FileContent.DATA.id()) {
+            return manifestContentId == ManifestContent.DATA.id();
+          } else {
+            return manifestContentId == ManifestContent.DELETES.id();
+          }
+        }
+        return ROWS_MIGHT_MATCH;
       }
 
       @Override
       public <T> Boolean notEq(BoundReference<T> ref, Literal<T> lit) {
-        return compareDateFileContent(ref, lit, compareResult -> compareResult != 0);
+        return ROWS_MIGHT_MATCH;
```

Comment:
I think we can still do neq, right?  We can still extract the eq logic to a separate method, and then negate it here?  Let me know if im mistaken though